### ISSUE TITLE
Investigate compression methods on the minikube.iso

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#6340]
The compression method to use is a trade-off between size and speed...

Size testing with Buildroot 2019.02:

```
498M	rootfs.cpio
191M	rootfs.cpio.lz4
165M	rootfs.cpio.gz
149M	rootfs.cpio.bz2
89M	rootfs.cpio.xz
```

Speed findings from Ubuntu 19.10:

```
940m    BZIP2
705m    XZ
239m    LZO
235m    GZIP
184m    LZ4
```

We could investigate changing to a slightly bigger ISO, that boots faster.

**Comment by afbjorklund** [comment]
Here is the boot time and rootfs size for minikube, before and after:

bzip2
```
[    0.283915] Unpacking initramfs...
[   19.837292] Freeing initrd memory: 144364K
```

gzip
```
[    0.281452] Unpacking initramfs...
[    2.928928] Freeing initrd memory: 160124K
```

So 6-7x times faster (16.9s), 10-11% bigger (15.4M)

**Comment by priyawadhwa** [comment]
@afbjorklund this is awesome. I did a similar analysis a few days ago and got these results:


Compression | Image Size | Time to Extract FS | Total minikube start | Buildroot package (from here)
-- | -- | -- | -- | --
bzip2 | 176 M | 17s | 232s | BR2_TARGET_ROOTFS_CPIO_BZIP2=y
gzip | 543M | 2.8s | 208s | BR2_TARGET_ROOTFS_CPIO_GZIP=y
lzma | 478M | 49.5 | 256s | BR2_TARGET_ROOTFS_CPIO_LZMA=y
No compression | 855M | 1.7s | 206s | BR2_TARGET_ROOTFS_CPIO_NONE=y
xz | 466M | 14.4s | 275s | BR2_TARGET_ROOTFS_CPIO_XZ
lz4 | 570M | 1.8s | 203s | BR2_TARGET_ROOTFS_CPIO_LZ4
Gzip / Anders | 543M | 2.9s | 208s | From PR #6341

The time to extract FS is about the same but I'm not sure why the size of my ISO is so much bigger. 

I'll rebuild the iso and see what I get, but if the size of gzip is only 235M I think that's a good compromise between size and speed.

**Comment by afbjorklund** [comment]
I used lz4 for the kernel and gzip for the rootfs, it seemed like a good trade-off (conservative upgrade).

Not sure if your size includes hardlinks or not (git has plenty!), my final minikube.iso size was 167M

**Comment by priyawadhwa** [comment]
Cool I rebuilt and got 164M for the changes in your PR -- looks good to me!

**Comment by tstromberg** [comment]
The trade-off between size & speed can be mitigated by distributing the ISO as a compressed file, and decompressing it once per ISO version post-download.

That said, the ideal form is one that works great for the first start and subsequent starts.

**Comment by afbjorklund** [comment]
e.g. CoreOS does this, but I'm not sure it is worth the documentation and/or extra step involved ?
And of course it will also use more disk space in cache, which in their case works out to 1.7 GB...

fedora-coreos-31.20200113.3.1-qemu.x86_64.qcow2.xz

Even if our ISO is "only" third (475M), that's still +300 MB ?

Some other systems even allow for running packages and images in their compressed form!
Unfortunately our limited technology (buildroot and docker) requires us to decompress it first.

I still occasionally run out of disk when using KVM2 driver.

I'm not really sure why that image runs off to 5G ? (#3155)

**Comment by afbjorklund** [comment]
Merged in `v1.7.0-beta.1~20`
